modules/renter/contractor/contractor.go

```diff
@@ -96,10 +113,14 @@ func (c *Contractor) ContractByID(id types.FileContractID) (modules.RenterContra
 // Contracts returns the contracts formed by the contractor in the current
 // allowance period. Only contracts formed with currently online hosts are
 // returned.
-func (c *Contractor) Contracts() (cs []modules.RenterContract) {
+func (c *Contractor) Contracts() []modules.RenterContract {
 	c.mu.RLock()
 	defer c.mu.RUnlock()
-	return c.onlineContracts()
+	cs := make([]modules.RenterContract, 0, len(c.contracts))
+	for _, contract := range c.contracts {
+		cs = append(cs, contract)
+	}
+	return cs
 }
```

Comment:
`Contracts()` now just returns all contracts, instead of merely returning the set of online contracts. The `GoodForUpload` and `GoodForRenew` fields. The uploader and downloader workers have timeouts for failed interactions, and contracts which are offline get marked at !GoodForUpload and !GoodForRenew both.

Though, I'm realizing now that there is no signal to the download workers if a contract is offline. Might need to add some more testing for that, though there are a lot of tests that perform downloads while hosts are offline or missing, and they all seem to be doing okay.